src/relations.jl

```diff
@@ -1212,6 +1237,42 @@ PhasePartition_equil(ts::AbstractPhaseNonEquil) = PhasePartition_equil(
     typeof(ts),
 )
 
+
+"""
+    PhasePartition_equil_given_p(param_set, T, p, q_tot, phase_type)
+Partition the phases in equilibrium, returning a [`PhasePartition`](@ref) object using the
+[`liquid_fraction`](@ref) function where
+ - `param_set` an `AbstractParameterSet`, see the [`Thermodynamics`](@ref) for more details
+ - `T` temperature
+ - `p` air pressure
+ - `q_tot` total specific humidity
+ - `phase_type` a thermodynamic state type
+The residual `q.tot - q.liq - q.ice` is the vapor specific humidity.
+"""
+function PhasePartition_equil_given_p(
+    param_set::APS,
+    T::FT,
+    p::FT,
+    q_tot::FT,
+    phase_type::Type{<:ThermodynamicState},
+) where {FT <: Real}
+
+    q_v_sat = q_vap_saturation_from_pressure(param_set, q_tot, p, T, phase_type)
+    _liquid_frac = liquid_fraction(param_set, T, phase_type)
+    q_c = q_tot - q_v_sat
+    q_liq = _liquid_frac * q_c
+    q_ice = (1 - _liquid_frac) * q_c
+    return PhasePartition(q_tot, q_liq, q_ice)
+end
+
+PhasePartition_given_p(ts::PhaseEquil) = PhasePartition_equil_given_p(
```

Comment:
This is the one big part of this PR that I'm uncomfortable with. This function should really be called `PhasePartition`-- but then that leaves us with a very opaque view of what's happening. Sometimes `PhasePartition_equil` is called, and other times `PhasePartition_equil_given_p` is called, and it's not clear from which constructors which route is taken. 

The way that we've got around this so far is by having one method call the other, and the method that implements the actual relation is the "core" method. This is really nice because it ensures that all relations we use are synchronized / consistent.

And the core problem with making `PhasePartition_equil_given_p` the core method is that, when pressure is _not_ given, we can't construct it when determining the phase partition (because it _depends_ on the phase partition).

I've discussed with @yairchn that, perhaps we can put a bit more effort into ensuring that we cannot make the existing methods work.

Reply:
Ok, I think we've resolved things. See [this comment](https://github.com/CliMA/Thermodynamics.jl/pull/52#issuecomment-930411505).